streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -0,0 +1,254 @@
+        }
+
+    }
+
+    /**
+     * Send a request to kafka broker of this client. Polls the request for a given number of iterations to receive the response.
+     *
+     * @param request
+     * @param apiKeys
+     * @param callback
+     */
+    public ClientResponse sendRequest(final Struct request, final ApiKeys apiKeys, final RequestCompletionHandler callback) {
+
+        Node brokerNode = kafkaClient.leastLoadedNode(new SystemTime().milliseconds());
+        String brokerId = Integer.toString(brokerNode.id());
+
+        SystemTime systemTime = new SystemTime();
+
+        RequestSend send = new RequestSend(brokerId,
+                kafkaClient.nextRequestHeader(apiKeys),
+                request);
+
+        ClientRequest clientRequest = new ClientRequest(systemTime.milliseconds(), true, send, callback);
+
+
+        final long timeout = systemTime.milliseconds() + MAX_WAIT_TIME_MS;
+        while (!kafkaClient.ready(brokerNode, systemTime.milliseconds()) && systemTime.milliseconds() < timeout) {
+            kafkaClient.poll(streamsConfig.getLong(StreamsConfig.POLL_MS_CONFIG), systemTime.milliseconds());
+        }
+        if (!kafkaClient.ready(brokerNode, systemTime.milliseconds())) {
+            throw new StreamsException("Request timeout.");
+        }
+        kafkaClient.send(clientRequest, systemTime.milliseconds());
+
+        final long timeout2 = systemTime.milliseconds() + MAX_WAIT_TIME_MS;
+        // Poll for the response.
+        while (systemTime.milliseconds() < timeout2) {
+            List<ClientResponse> responseList = kafkaClient.poll(streamsConfig.getLong(StreamsConfig.POLL_MS_CONFIG), systemTime.milliseconds());
+            for (ClientResponse clientResponse: responseList) {
+                if (clientResponse.request().equals(clientRequest)) {
+                    return clientResponse;
+                }
+            }
+        }
+        throw new StreamsException("StreamsKafkaClient failed to send the request: " + apiKeys.name());
+    }
+
+
+    /**
+     * Get the metadata for a topic.
+     * @param topic
+     * @return
+     */
+    public MetadataResponse.TopicMetadata getTopicMetadata(final String topic) {
+        MetadataRequest metadataRequest = new MetadataRequest(Arrays.asList(topic));
+
+        RequestCompletionHandler callback = new RequestCompletionHandler() {
+            public void onComplete(ClientResponse response) {
+                // Do nothing!
+            }
+        };
+
+        ClientResponse clientResponse = sendRequest(metadataRequest.toStruct(), ApiKeys.METADATA, callback);
+        MetadataResponse metadataResponse = new MetadataResponse(clientResponse.responseBody());
+
+        for (MetadataResponse.TopicMetadata topicMetadata: metadataResponse.topicMetadata()) {
+            if (topicMetadata.topic().equalsIgnoreCase(topic)) {
+                if (topicMetadata.error().code() == 0) {
```

Comment:
See my comment above in `InternalTopicManager`; we can probably just return the matching `MetadataResponse.TopicMetadata` instance here. As far as i know it will never be null.
If we don't find one then we should probably throw an exception